src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -149,11 +194,49 @@ private String getLocalName(Local local) {
     }
 
     /*
-     * Returns a Value containing the name
+     * Returns a String containing the name
      * of the variable used to assign an array;
      * e.g:  int[] arr = {0, 1}; --> "arr"
      */
-    private Value getJArrayRefName(ValueBox valueBox) {
-        return (((JArrayRef) valueBox.getValue()).getBaseBox().getValue());
+    private String getJArrayRefName(JArrayRef jArrayRef) {
+        return jArrayRef.getBaseBox().getValue().toString();
+    }
+
+    /*
+     * Returns a String containing the name
+     * of the variable used to assign a static variable;
+     * e.g:  private static int x; --> "x"
+     */
+    private String getStaticFieldRefName(StaticFieldRef staticFieldRef) {
+        return staticFieldRef.getField().getName();
+    }
+
+    /*
+     * Returns a list of Boxes containing Values ​​defined
+     * in this Unit for the types selected in the filter.
+     */
+    private Stream<ValueBox> getStatementAssociatedWithUnit(Unit u) {
+        return u.getDefBoxes()
+                .stream()
+                .filter(valueBox -> valueBox.getValue() instanceof Local
+                        || valueBox.getValue() instanceof JArrayRef
+                        || valueBox.getValue() instanceof StaticFieldRef
+                );
+    }
+
+    private boolean isLeftStatement(Unit u) {
```

Comment:
These methods should be in the auxiliary methods of ReachDefinitionAnalysis, because others classes can use them.

Reply:
Done!